README.md

```diff
@@ -9,3 +9,23 @@ Currently causal language models and sequence-to-sequence models are supported.
 #### Notes
 
 - The data model for text generative capabilities is baked into this repository itself at `caikit_nlp/data_model/generation.py`.
+
+## Development guidelines
+
+To contribute code to this repository, kindly fork the repository and create [pull requests](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request-from-a-fork) with your contribution. For the contribution to be reviewed, we require unit tests for every new functionality added. We have some checks in place for code formatting and linting. We recommend that contributors ensure checks pass locally before opening pull requests for review. The following commands can be used for assistance:
+
+### Useful commands
```

Comment:
Hi @hickeyma, currently we are missing this type of information from our projects, what would be the best way you would recommend to get this documented? 

Shall we start with adding this information in the `README` ?